src/NuGet.Clients/VsExtension/NuGetPackage.cs

```diff
@@ -196,12 +196,30 @@ private ISolutionManager SolutionManager
                 if (_solutionManager == null)
                 {
                     _solutionManager = ServiceLocator.GetInstance<ISolutionManager>();
+                    _solutionManager.AfterNuGetProjectRenamed += SolutionManager_NuGetProjectRenamed;
```

Comment:
PackageManagerControl doesn't have the necessary context (It needs to call FindExistingWindowFrame in NugetPackage.cs), that's why I decided to add it in NugetPackage.cs itself.